kernel/yosys.cc

```diff
@@ -318,8 +318,11 @@ int run_command(const std::string &command, std::function<void(const std::string
 #else
 	return WEXITSTATUS(ret);
 #endif
-}
+#else
+	log_warning("Shell command execution is disabled in this build.\n");
```

Comment:
This should definitely not be a warning but an error.

Reply:
It signifies an error through the return value, the warning is a courtesy to help the user understand the cause of the error.

The outside code needs handling of an error anyway for platforms with spawning enabled, and relying on that has the advantage that the handling may be log_cmd_error, which is better for user experience. We can’t do blanket log_cmd_error from within the run_command function.

Reply:
I removed it from the PR